api/Business/BusinessObjects/SensorBusinessObject.cs

```diff
@@ -47,6 +51,59 @@ public async Task<OperationResult> CreateSensor(Sensor record)
             });
         }
 
+        public async Task<OperationResult<List<SensorBusinessModel>>> ListSensors()
+        {
+            return await ExecuteOperation(async () =>
+            {
+                string currentUser = _httpContextAccessor.HttpContext!.User.FindFirst(ClaimTypes.NameIdentifier)!.Value;
+                long currentUserId = long.Parse(currentUser);
+
+                if (currentUser == null)
+                {
+                    throw new Exception("Session expired");
+                }
+
+                List<Sensor> sensors = await _sensorDataAccessObject.ListSensors(currentUserId);
+                List<SensorBusinessModel> result = sensors.Select(s => new SensorBusinessModel(s)).ToList();
```

Comment:
does this works if the sensors result is null?

Reply:
was missing that, fixed now